File: apps/go-example/calculator_story_test.go
package example

import (
	"testing"

	es "github.com/anthropics/executable-stories/packages/executable-stories-go"
)

func TestCalculatorAddsTwoNumbers(t *testing.T) {
	s := es.Init(t, "Calculator adds two numbers")
	s.Given("two numbers 5 and 3")
	a, b := 5, 3
	s.When("the numbers are added")
	result := Add(a, b)
	s.Then("the result is 8")
	if result != 8 {
		t.Fatalf("expected 8, got %d", result)
	}
}

func TestCalculatorSubtractsTwoNumbers(t *testing.T) {
	s := es.Init(t, "Calculator subtracts two numbers")
	s.Given("two numbers 10 and 4")
	a, b := 10, 4
	s.When("the second is subtracted from the first")
	result := Subtract(a, b)
	s.Then("the result is 6")
	if result != 6 {
		t.Fatalf("expected 6, got %d", result)
	}
}

func TestCalculatorMultipliesTwoNumbers(t *testing.T) {
	s := es.Init(t, "Calculator multiplies two numbers")
	s.Given("two numbers 7 and 6")
	a, b := 7, 6
	s.Note("This is a note")
	s.When("the numbers are multiplied")
	result := Multiply(a, b)
	s.Then("the result is 42")
	if result != 42 {
		t.Fatalf("expected 42, got %d", result)
	}
}

func TestCalculatorDividesTwoNumbers(t *testing.T) {
	s := es.Init(t, "Calculator divides two numbers")
	s.Given("two numbers 20 and 4")
	a, b := 20, 4
	s.When("the first is divided by the second")
	result := Divide(a, b)
	s.Then("the result is 5")
	if result != 5 {
		t.Fatalf("expected 5, got %d", result)
	}
}

func TestCalculatorThrowsOnDivisionByZero(t *testing.T) {
	s := es.Init(t, "Calculator throws error on division by zero")
	s.Note("Division by zero should throw an error")
	s.Given("a number 10 and zero")
	a, b := 10, 0
	s.When("division is attempted")
	var panicked bool
	func() {
		defer func() {
			if recover() != nil {
				panicked = true
			}
		}()
		_ = Divide(a, b)
	}()
	if !panicked {
		t.Fatal("expected Divide(10, 0) to panic")
	}
	s.Then("an error is thrown")
}

File: apps/jest-example/src/calculator.story.test.ts
import { describe, expect, it } from '@jest/globals';
import { story } from 'executable-stories-jest';
import { add, divide, multiply, subtract } from './calculator.js';

describe('Calculator', () => {
  it('Calculator adds two numbers', () => {
    story.init();

    story.given('two numbers 5 and 3');
    const a = 5;
    const b = 3;

    story.when('the numbers are added');
    const result = add(a, b);

    story.then('the result is 8');
    expect(result).toBe(8);
  });

  it('Calculator subtracts two numbers', () => {
    story.init();

    story.given('two numbers 10 and 4');
    const a = 10;
    const b = 4;

    story.when('the second is subtracted from the first');
    const result = subtract(a, b);

    story.then('the result is 6');
    expect(result).toBe(6);
  });

  it('Calculator multiplies two numbers', () => {
    story.init();

    story.given('two numbers 7 and 6');
    const a = 7;
    const b = 6;

    story.when('the numbers are multiplied');
    const result = multiply(a, b);

    story.then('the result is 42');
    expect(result).toBe(42);
  });

  it('Calculator divides two numbers', () => {
    story.init();

    story.given('two numbers 20 and 4');
    const a = 20;
    const b = 4;

    story.when('the first is divided by the second');
    const result = divide(a, b);

    story.then('the result is 5');
    expect(result).toBe(5);
  });

  it('Calculator throws error on division by zero', () => {
    story.init();

    story.note('Division by zero should throw an error');

    story.given('a number 10 and zero');
    const a = 10;
    const b = 0;
    let error: Error | null = null;

    story.when('division is attempted');
    try {
      divide(a, b);
    } catch (e) {
      error = e as Error;
    }

    story.then('an error is thrown');
    expect(error).not.toBeNull();
    expect(error?.message).toBe('Cannot divide by zero');
  });
});
